presto-hudi/src/main/java/com/facebook/presto/hudi/HudiModule.java

```diff
@@ -132,6 +138,34 @@ public ExecutorService createCachingHiveMetastoreExecutor(MetastoreClientConfig
                 daemonThreadsNamed("hive-metastore-hudi-%s"));
     }
 
+    @ForHudiSplitAsyncQueue
+    @Singleton
+    @Provides
+    public ExecutorService createHudiSplitManagerExecutor()
+    {
+        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%s"));
+    }
+
+    @ForHudiSplitSource
+    @Singleton
+    @Provides
+    public ScheduledExecutorService createSplitLoaderExecutor(HudiConfig hudiConfig)
```

Comment:
is this created per query? whats the lifecycle for this executor service.

Reply:
All these executor services are there for the whole session till the server is alive. They are not created per query.